src/Features/CSharp/Portable/CodeLens/CSharpCodeLensDisplayInfoService.cs

```diff
@@ -0,0 +1,202 @@
+
+                case SyntaxKind.IndexerDeclaration:
+                    // The name will be "namespace.class.this[type] - we want "namespace.class[type] Indexer"
+                    symbol = semanticModel.GetDeclaredSymbol(node);
+                    displayName = symbol.ToDisplayString(symbolDisplayFormat);
+                    var openBracketIndex = displayName.IndexOf('[');
+                    var lastDotBeforeOpenBracketIndex = displayName.LastIndexOf('.', openBracketIndex, openBracketIndex);
+
+                    displayName = displayName.Substring(0, lastDotBeforeOpenBracketIndex) +
+                                         displayName.Substring(openBracketIndex) +
+                                         " Indexer";
+                    break;
+
+                case SyntaxKind.OperatorDeclaration:
+                    // The name will be "namespace.class.operator +(type)" - we want namespace.class.+(type) Operator
+                    symbol = semanticModel.GetDeclaredSymbol(node);
+                    displayName = symbol.ToDisplayString(symbolDisplayFormat);
+                    var spaceIndex = displayName.IndexOf(' ');
+                    lastDotBeforeSpaceIndex = displayName.LastIndexOf('.', spaceIndex, spaceIndex);
+
+                    displayName = displayName.Substring(0, lastDotBeforeSpaceIndex + 1) +
+                                         displayName.Substring(spaceIndex + 1) +
+                                         " Operator";
+                    break;
+
+                case SyntaxKind.ConversionOperatorDeclaration:
+                    // The name will be "namespace.class.operator +(type)" - we want namespace.class.+(type) Operator
+                    symbol = semanticModel.GetDeclaredSymbol(node);
+                    displayName = symbol.ToDisplayString(symbolDisplayFormat);
+                    var firstSpaceIndex = displayName.IndexOf(' ');
+                    var secondSpaceIndex = displayName.IndexOf(' ', firstSpaceIndex + 1);
+                    lastDotBeforeSpaceIndex = displayName.LastIndexOf('.', firstSpaceIndex, firstSpaceIndex);
+
+                    displayName = displayName.Substring(0, lastDotBeforeSpaceIndex + 1) +
+                                         displayName.Substring(secondSpaceIndex + 1) +
+                                         " Operator";
+                    break;
+
+                case SyntaxKind.UsingDirective:
+                    // We want to see usings formatted as simply "Using", prefaced by the namespace they are in
+                    enclosingScopeString = GetEnclosingScopeString(node, semanticModel, symbolDisplayFormat);
+                    displayName = string.IsNullOrEmpty(enclosingScopeString) ? "Using" : enclosingScopeString + " Using";
+                    break;
+
+                case SyntaxKind.ExternAliasDirective:
+                    // We want to see aliases formatted as "Alias", prefaced by their enclosing scope, if any
```

Comment:
It looks like what you get is the enclosing namespace, so "namespace NS1 { external alias foo; }" would produce a display string for foo of "NS1 Alias." I'm not sure whether this code path is really exercised in the CodeLens reference scenario, I will check with the Code Lens team.

---
In reply to: [74156763](https://github.com/dotnet/roslyn/pull/13037#discussion_r74156763) [](ancestors = 74156763)